### DISCUSSION TITLE
Will it reduce RSS memory when share ByteBufAllocator

### DISCUSSION THREAD

**Discussion by He-Pin** [netty/netty#14556] [General]
Hi, Will it reduce RSS Memory usage when an application sharing/using the `PooledByteBufAllocator.DEFAULT`/shared one, instead of creating multi `ByteBufAllocator` instances?

I'm asking because we are using class loader isolation, where several libraries have their own `io.netty.*`, as many `ByteBufAllocator` instances.

**Comment by normanmaurer** [comment]
Sure if you share the same instance it will reduce the overhead. That said with class loader isolation I think this is not true as every class loader would have their own static one.

**Reply by He-Pin** [reply]
Thanks for the reply.
Yes, that's why we are going to make the`io.netty.*` shared across our plugins, only visible to our plugins but never exported to users code.
<img width="657" alt="image" src="https://github.com/user-attachments/assets/a19c1f30-efe2-4d0f-8e78-aaeda03ecda2" />

**Reply by He-Pin** [reply]
<img width="1312" alt="image" src="https://github.com/user-attachments/assets/851bb2b8-0df1-43b1-b07a-6e52bcb1e898" />
It works @normanmaurer , we still need some polish before rolling it out

**Comment by He-Pin** [comment]
<img width="580" alt="image" src="https://github.com/user-attachments/assets/9cbbf5c6-5536-4a86-9f3e-54dbbfaeb2dc" />

We are betaing, but the RT reduced after sharing the bytebuf, is that something related to CPU Cache?
```
$lscpu | grep -i cache
L1d cache:             32K
L1i cache:             32K
L2 cache:              1024K
L3 cache:              36608K
```

@franz1981 Do you have any idea?

**Reply by franz1981** [reply]
Having less distinct memory alive still means that you can fit better into some of the caches while dealing with it, but the RSS usage is more related the actually requested to the OS and which is virtually mapped and backed by actual physical memory (because turn to be physically accessed/modified/used).

**Reply by He-Pin** [reply]
<img width="575" alt="image" src="https://github.com/user-attachments/assets/86eafe48-1973-4343-9430-6d5ea029754f" />
<img width="575" alt="image" src="https://github.com/user-attachments/assets/a3d39ca6-51a9-412c-863a-110c89451a69" />

That's an unexpected surprise.

**Reply by franz1981** [reply]
happy that this is working better than expected than :)
